src/commissioner/ExampleCommissioningStateMachine.h

```diff
@@ -0,0 +1,664 @@
+    }
+    auto CreateNetworkAdded(NetworkId networkId)
+    {
+        return State::Create<NetworkAdded>(mCtx, mCommissionee, networkId);
+    }
+    auto CreateInvokingConnectNetwork(NetworkId networkId)
+    {
+        return State::Create<InvokingConnectNetwork>(mCtx, mCommissionee, networkId);
+    }
+    auto CreateNetworkEnabled()
+    {
+        return State::Create<NetworkEnabled>(mCtx, mCommissionee);
+    }
+    auto CreateOperationalDiscovery()
+    {
+        return State::Create<OperationalDiscovery>(mCtx, mCommissionee);
+    }
+    auto CreateInitiatingCase(OperationalRecord record)
+    {
+        return State::Create<InitiatingCase>(mCtx, mCommissionee, record);
+    }
+    auto CreateCaseComplete()
+    {
+        return State::Create<CaseComplete>(mCtx, mCommissionee);
+    }
+    auto CreateInvokingCommissioningComplete()
+    {
+        return State::Create<InvokingCommissioningComplete>(mCtx, mCommissionee);
+    }
+    auto CreateCommissioningComplete()
+    {
+        return State::Create<CommissioningComplete>(mCtx, mCommissionee, mOnSuccess);
+    }
+    auto CreateFailed()
+    {
+        return State::Create<Failed>(mCtx, mCommissionee, mOnFailure);
+    }
+    // clang-format on
+
+private:
+    Context & mCtx;
+    Commissionee & mCommissionee;
+    OpCredsIssuer * mIssuer;
+    FabricIndex mFabricIndex = kUndefinedFabricIndex;
+    NodeId mNodeId           = kUndefinedNodeId;
+    ByteSpan mOperationalDataset;
+    ByteSpan mSsid;
+    ByteSpan mWiFiCredentials;
+    OnSuccess mOnSuccess;
+    OnFailure mOnFailure;
+};
+
+struct Transitions
+{
+    Transitions(Context & ctx, Commissionee & commissionee) : mFactory(ctx, commissionee), mCtx(ctx) {}
+    chip::StateMachine::Optional<State> operator()(const State & state, const Event & event);
+    State GetInitState() { return mFactory.CreateIdle(); }
+    StateFactory mFactory;
+    Context & mCtx;
+};
+
+class ExampleCommissioningStateMachine : public chip::StateMachine::StateMachine<Context, State, Event, Transitions>
+{
+
+public:
+    ExampleCommissioningStateMachine() :
+        chip::StateMachine::StateMachine<Context, State, Event, Transitions>(mTransitions), mTransitions(*this, mCommissionee)
+    {}
+    void Init(SystemState * systemState, OpCredsIssuer * issuer, FabricIndex fabricIndex, NodeId nodeId);
+    void Init(SystemState * systemState, OpCredsIssuer * issuer, FabricIndex fabricIndex, NodeId nodeId,
+              ByteSpan operationalDataset, ByteSpan ssid, ByteSpan wiFiCredentials);
+    void Shutdown();
+    CHIP_ERROR Commission(const char * onboardingPayload, OnSuccess onSuccess, OnFailure onFailure);
+    CHIP_ERROR Commission(chip::SetupPayload & onboardingPayload, OnSuccess onSuccess, OnFailure onFailure);
+    CHIP_ERROR Commission(chip::RendezvousInformationFlags flags, uint16_t discriminator, uint32_t setUpPINCode,
+                          OnSuccess onSuccess, OnFailure onFailure);
```

Comment:
A comment describing which method to use under different circumstances would help.